tests/conftest.py

```diff
@@ -0,0 +1,221 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from contextlib import contextmanager
+import io
+import json
+import os
+import random
+import sys
+import uuid
+
+import boto3
+from botocore.client import Config as Boto3Config
+from moto import mock_s3
+import requests_mock
+import pytest
+
+
+# Insert build/ directory in sys.path so we can import submitter
+sys.path.insert(
+    0,
+    os.path.join(
+        os.path.dirname(os.path.dirname(__file__)),
+        'build'
+    )
+)
+
+
+from submitter import CONFIG, generate_s3_key, handler  # noqa
+
+
+class LambdaContext:
+    """Context class that mimics the AWS Lambda context
+
+    http://docs.aws.amazon.com/lambda/latest/dg/python-context-object.html
+
+    """
+    def __init__(self):
+        self.aws_request_id = uuid.uuid4().hex
+
+        self.log_group_name = '/aws/lambda/test'
+        self.log_stream_name = '2016-11-15blahblah'
+
+        self.function_name = 'test'
+        self.memory_limit_in_mb = '384'
+        self.function_version = '1'
+        self.invoked_function_arn = 'arn:aws:lambda:us-west-2:blahblah:function:test'
+
+        # FIXME(willkg): Keeping these as None until we need them.
+        self.client_context = None
+        self.identity = None
+
+    def get_remaining_time_in_millis(self):
+        # FIXME(willkg): Implement this when we need it
+        return 5000
+
+
+class SubmitterClient:
+    """Class for submitter in the AWS lambda environment"""
+    def crash_id_to_key(self, crash_id):
+        return 'v2/raw_crash/{entropy}/{date}/{crashid}'.format(
+            entropy=crash_id[0:3],
+            date='20' + crash_id[-6:],
+            crashid=crash_id
+        )
+
+    def build_crash_save_events(self, keys):
+        if isinstance(keys, str):
+            keys = [keys]
+
+        # FIXME(willkg): This only generates a record that has the stuff that
+        # submitter is looking for. It's not a full record.
+        return {
+            'Records': [
+                {
+                    'eventSource': 'aws:s3',
+                    'eventName': 'ObjectCreated:Put',
+                    's3': {
+                        'object': {
+                            'key': key
+                        },
+                        'bucket': {
+                            'name': 'dev_bucket'
+                        }
+                    }
+                }
+                for key in keys
+            ]
+        }
+
+    def run(self, events):
+        result = handler(events, LambdaContext())
+        return result
+
+
+@pytest.fixture
+def client():
```

Comment:
This seems less of a fixture and more of three utility methods grouped together, since there's no real state shared between them. Should we split them up?

Reply:
In this case, the client is effectively the AWS Lambda runtime. It has three functions that are all needed to invoke the Lambda function. I don't think it helps at all to split it up.

I can change the name to something like "runtime" or something like that. I used "client" because it's a similar role to the client used for testing Django apps.